### DISCUSSION TITLE
Automatic columns for tagboards with subtags

### DISCUSSION THREAD

**Discussion by naxey** [roovo/obsidian-card-board#173] [Ideas]
This feature would automatically create a column in the tagboard named after the marked section in the nested tag. It would also allow for a negative prompt to specify tags that should **not** auto-generate lists in the tagboard.

The proposed settings naming scheme for automatic population:
<img width="789" alt="image" src="https://github.com/roovo/obsidian-card-board/assets/40920984/932c9da0-ea65-4c84-a816-f517d2030e7f">

These settings would create columns for any nested tags with the parent tag `#parent-tag/...` but not the tag `#parent-tag/ignored-tag` or the tag `#parent-tag` without nesting.

Using these settings, the tag `#parent-tag/suggestions` would create the column `suggestions` but the tag `#parent-tag/ignored-tag` would not create a column.

For reference this is the official naming scheme for tags:
```
You can use any of the following characters in your tags:

Alphabetical letters
Numbers
Underscore (_)
Hyphen (-)
Forward slash (/) for Nested tags
Tags must contain at least one non-numerical character.
```

**Comment by roovo** [comment]
This is certainly something I have thought about supporting!  It would have worked really nicely with the way boards used to be defined as (as you have said) it would be an option for a special type of tag board.  Now I have moved away from defining boards in this way (a board is now more of a collection of separately defined columns) this fits in less well (the main reason I moved to a board being a collection of separate columns was for easy tracking of their order for drag and drop).

I can still think of ways of doing this, but they are all a bit more awkward than they were, both conceptually and in implementation - hah, progress!  

Before this change, there were some aspects I never bottomed out, like what to do if a tag/subtag combination is removed from the vault (because all lines with it on are deleted), should I remove the column from the board.  Should I only look at tasks when picking the tag/sub-tag combinations or should I look at the complete vault?

I am definitely still mulling this over!

**Reply by myndatlas** [reply]
While I appreciate the drag and drop, I really miss the ability to rapidly copy/paste the same status columns to a new board with the old method. It's so much more tedious now.

**Reply by roovo** [reply]
I hear you - one thing that might help (when I get to implemting it) could be the ability to duplicte boards - #196.  Although I do agree that the ability to quickly type some tags into a text input was massively convenient - will keep this in mind as I develop things further!
